### DISCUSSION TITLE
Pin.irq() keyword args not accepted - Cannot wake from deepsleep with GPIO falling edge

### DISCUSSION THREAD

**Discussion by davidmcnabnz** [micropython/micropython#11230] [RP2040 / Pico]
Hi,

With the Pi Pico W, I want to enter deep sleep (aka 'dormant'), where the device wakes when it sees a falling edge on a given pin.

Pin.irq() documentation gives:
```
Pin.irq(handler=None, trigger=Pin.IRQ_FALLING | Pin.IRQ_RISING, *, priority=1, wake=None, hard=False)
```
For the RP2040 port, the Pin.irq() method does not accept the 'wake' and 'hard' keywords. I was hoping to use these to enable the interrupt handler to wake the Pico.

Also, the constants machine.DEEPSLEEP, machine.SLEEP And machine.IDLE are not defined in the RP2040 port.

Can anyone offer any advice, for getting a Pico to wake out of deep sleep with a falling-edge GPIO input?

Cheers
David

**Comment by Josverl** [comment]
Perhaps the  constants from the stm32 can be used?

https://github.com/Josverl/micropython-stubs/blob/main/stubs/micropython-v1_19_1-stm32/machine.py#L8-L12

**Comment by wcndyd** [comment]
bro, i want to know whether you solve this question or not, I meet this problem  too

**Reply by davidmcnabnz** [reply]
My solution is some added circuitry to implement a basic logic latch plus programmable timer. The Pico then drives a pin to power itself off completely.

Then, it can get powered back on at regular intervals by the timer, or powered on as needed by a sensor.

Once powered on, the Pico does what it needs (eg WiFi connection, send an alarm, or just check on things) before powering itself back off.

The parts for the gate latch and timer are MOSFET based, so only consume a few nanoAmps while in standby.

**Reply by wcndyd** [reply]
thank you for your reply. it's a great idea, but my problem is that I want define interrupt priority .Every time when I want use this function, it always appear errors like your situation.  I don't kwon how to deal with Micropython , maybe I should change my methods as well as my code
